Sming/sming.mk

```diff
@@ -0,0 +1,662 @@
+#
+#==============================================================================
+#   user settings
+#------------------------------------------------------------------------------
+#   your flash memory size (required)
+ifeq ("$(SPI_SIZE)","")
+    $(error please set SPI_SIZE)
+endif
+
+#   enable rboot (default: 0)
+RBOOT_ENABLED   ?= 0
+
+#   your source directories (default: app)
+MODULES         ?= app
+
+#   your extra include directies (default: none)
+EXTRA_INCDIR    ?=
+
+#   your extra libraries (default: none)
+EXTRA_LIBS      ?=
+
+#   your extra source files (default: none)
+EXTRA_SRC       ?=
+
+#   your spiffs size (default: 0)
+SPIFF_SIZE      ?= 0
+
+#   your source directory for spiffs files (default: files)
+SPIFF_FILES     ?= files
+
+#   your reserved flash area for user data (default: 0)
+USER_FLASH_SIZE ?= 0
+
+#   patch init data for system_get_vdd33 (default: 0)
+GET_VDD33       ?= 0
+
+#==============================================================================
+#   load os specific settings
+#------------------------------------------------------------------------------
+ifeq ($(OS),Windows_NT)
+	include $(SMING_HOME)/Windows.mk
+else
+	UNAME := $(shell uname -s)
+	include $(SMING_HOME)/$(UNAME).mk
+endif
+
+#==============================================================================
+#   default serial settings
+#------------------------------------------------------------------------------
+#   use the maximum speed supported by your board to get things done faster
+COM_SPEED_ESPTOOL   ?= 921600
+
+#   use bootloader default (74880) to see bootloader messages
+COM_SPEED_SERIAL    ?= 74880
+
+DEFINES += COM_SPEED_SERIAL
+
+#==============================================================================
+#   path settings
+#------------------------------------------------------------------------------
+BUILD_BASE      = out/build
+FW_BASE         = out/firmware
+
+#   SDK directories
+SDK_BASE        ?= $(ESP_HOME)/sdk
+SDK_TOOLS       ?= $(SDK_BASE)/tools
+SDK_ROMS        := $(SDK_BASE)/bin
+SDK_LIBDIR      := $(SDK_BASE)/lib
+SDK_INCDIR      := $(SDK_BASE)/include
+
+XTENSA_TOOLS    := $(ESP_HOME)/xtensa-lx106-elf/bin
+
+USER_LIBDIR     := $(SMING_HOME)/compiler/lib/
+THIRD_PARTY_DIR := $(SMING_HOME)/third-party
+
+#==============================================================================
+#   tools
+#------------------------------------------------------------------------------
+ESPTOOL2        := esptool2
+SPIFFY          := $(SMING_HOME)/spiffy/spiffy
+
+CC              := $(XTENSA_TOOLS)/xtensa-lx106-elf-gcc
+CXX             := $(XTENSA_TOOLS)/xtensa-lx106-elf-g++
+AR              := $(XTENSA_TOOLS)/xtensa-lx106-elf-ar
+LD              := $(XTENSA_TOOLS)/xtensa-lx106-elf-gcc
+OBJCOPY         := $(XTENSA_TOOLS)/xtensa-lx106-elf-objcopy
+OBJDUMP         := $(XTENSA_TOOLS)/xtensa-lx106-elf-objdump
+STRIP           := $(XTENSA_TOOLS)/xtensa-lx106-elf-strip
+
+PERL            ?= perl
```

Comment:
Get rid of Perl. Use stock bash commands and if bash cannot help only then use python.